### PR TITLE
Fix guids handling

### PR DESCRIPTION
**Pull request by zphang**

### CODE REVIEW

File: jiant/tasks/evaluate/core.py [hunk 1]
```diff
@@ -73,7 +73,7 @@ def update(self, batch_logits, batch_loss, batch, batch_metadata):
         self.logits_list.append(batch_logits)
         batch_guid = batch_metadata.get("guid")
         if batch_guid is not None:
-            self.guid_list.extend(batch_guid)
+            self.guid_list.append(batch_guid)
```
Comment (jeswan) [inline jiant/tasks/evaluate/core.py:76]: @HaokunLiu changed this line a few days ago. Did this change result in a bug?

Comment (jeswan) [reply jiant/tasks/evaluate/core.py:76]: https://github.com/nyu-mll/jiant/pull/1197

Comment (zphang) [reply jiant/tasks/evaluate/core.py:76]: Yes. `self.guid_list` should be a list of lists (which gets converted to an array with `np.concatenate`), and Haokun's change leads to an error when writing predictions.
Let me think about how best to test this.

Comment (jeswan) [reply jiant/tasks/evaluate/core.py:76]: @HaokunLiu what error were you seeing which caused you to change this line?

Comment (HaokunLiu) [reply jiant/tasks/evaluate/core.py:76]: Sorry, I misunderstood the design of this variable. 
Could you replace guid = accumulator.guid_list in MCTACOEvaluationScheme with guid = accumulator.get_guids() ?

**Review by jeswan — APPROVED** [review]

**Pull request merged**